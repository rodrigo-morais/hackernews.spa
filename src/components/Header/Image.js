import styled from 'styled-components'

export default styled.img`
	width: 18px;
	height: 18px;
	border: 1px #ffffff solid;
`
