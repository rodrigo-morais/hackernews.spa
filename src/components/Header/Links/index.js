import React from 'react'
import styled from 'styled-components'

import { HN_URL } from '../../../constants'

import Title from './Title'
import Link from './Link'

const Td = styled.td`
	line-height: 12pt;
	height: 10px;
	font-family: Verdana, Geneva, sans-serif;
	font-size: 10pt;
	color: #222222;
`
const links = [
	{ href: 'newest', text: 'new' },
	{ href: `${HN_URL}newcomments`, text: 'comments' },
	{ href: `${HN_URL}show`, text: 'show' },
	{ href: `${HN_URL}ask`, text: 'ask' },
	{ href: `${HN_URL}jobs`, text: 'jobs' },
	{ href: `${HN_URL}submit`, text: 'submit', last: true },
]

const Links = () => (
	<Td>
		<Title />
		{links.map((link, index) => (
			<Link key={index} data={link} />
		))}
	</Td>
)

export default Links
