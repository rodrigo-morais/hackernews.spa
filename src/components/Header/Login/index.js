import React from 'react'
import styled from 'styled-components'

import { HN_URL } from '../../../constants'

import A from '../../shared/A'

const Td = styled.td`
	line-height: 12pt;
	height: 10px;
	font-family: Verdana, Geneva, sans-serif;
	font-size: 10pt;
	color: #222222;
	text-align:right;
	padding-right:4px;
`

const Login = () => (
	<Td>
		<A href={`${HN_URL}login?goto=news`}>login</A>
	</Td>
)

export default Login
