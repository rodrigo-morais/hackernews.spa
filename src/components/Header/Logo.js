import React from 'react'
import styled from 'styled-components'

import A from '../shared/A'
import Image from './Image'

import logo from '../../../assets/images/logo.gif'

const Td = styled.td`
	width: 18px;
	height: 18px;
	padding: 0;
	padding-right: 4px;
`

const Logo = () => (
	<Td>
		<A href="#">
			<Image src={logo} />
		</A>
	</Td>
)

export default Logo
