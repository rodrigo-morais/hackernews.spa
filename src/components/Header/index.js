import React from 'react'
import styled from 'styled-components'

import Logo from './Logo'
import Links from './Links'
import Login from './Login'

import Tbody from '../shared/Tbody'
import Tr from '../shared/Tr'

const Table = styled.table`
	border: 0;
  width: 100%;
  padding: 2px;
`

const Td = styled.td`
	background-color: #ff6600;
`

const Header = () => (
	<tr>
		<Td>
			<Table>
				<Tbody>
					<Tr>
						<Logo />
						<Links />
						<Login />
					</Tr>
				</Tbody>
			</Table>
		</Td>
	</tr>
)

export default Header
