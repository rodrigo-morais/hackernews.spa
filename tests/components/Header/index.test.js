import React from 'react'
import renderer from 'react-test-renderer'

import Header from '../../../src/components/Header'

describe('Header', () => {
  describe('Rendering', () => {
    test('renders according to design', () => {
      const component = renderer.create(
				<Header />
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
