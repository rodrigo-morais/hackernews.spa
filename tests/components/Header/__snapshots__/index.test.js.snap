// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering renders according to design 1`] = `
<tr>
  <td
    className="sc-gZMcBi dKVBSg"
  >
    <table
      className="sc-iwsKbI dtWENv"
    >
      <tbody
        className="sc-htoDjs keqScR"
      >
        <tr
          className="sc-dnqmqq hJxOZJ"
        >
          <td
            className="sc-htpNat dvyfjR"
          >
            <a
              className="sc-bdVaJa gASuaN"
              href="#"
            >
              <img
                className="sc-bwzfXH gKcczb"
                src="test-file-stub"
              />
            </a>
          </td>
          <td
            className="sc-bZQynM kjNPar"
          >
            <b
              className="sc-bxivhb CxaZp"
            >
              <a
                className="sc-bdVaJa gASuaN"
                href="#"
              >
                Hacker News
              </a>
            </b>
            <div
              className="sc-ifAKCX dzurso"
            >
              <a
                className="sc-bdVaJa gASuaN"
                href="newest"
              >
                new
              </a>
              <span
                className="sc-EHOje ljUHpj"
              >
                |
              </span>
            </div>
            <div
              className="sc-ifAKCX dzurso"
            >
              <a
                className="sc-bdVaJa gASuaN"
                href="https://news.ycombinator.com/newcomments"
              >
                comments
              </a>
              <span
                className="sc-EHOje ljUHpj"
              >
                |
              </span>
            </div>
            <div
              className="sc-ifAKCX dzurso"
            >
              <a
                className="sc-bdVaJa gASuaN"
                href="https://news.ycombinator.com/show"
              >
                show
              </a>
              <span
                className="sc-EHOje ljUHpj"
              >
                |
              </span>
            </div>
            <div
              className="sc-ifAKCX dzurso"
            >
              <a
                className="sc-bdVaJa gASuaN"
                href="https://news.ycombinator.com/ask"
              >
                ask
              </a>
              <span
                className="sc-EHOje ljUHpj"
              >
                |
              </span>
            </div>
            <div
              className="sc-ifAKCX dzurso"
            >
              <a
                className="sc-bdVaJa gASuaN"
                href="https://news.ycombinator.com/jobs"
              >
                jobs
              </a>
              <span
                className="sc-EHOje ljUHpj"
              >
                |
              </span>
            </div>
            <div
              className="sc-ifAKCX dzurso"
            >
              <a
                className="sc-bdVaJa gASuaN"
                href="https://news.ycombinator.com/submit"
              >
                submit
              </a>
            </div>
          </td>
          <td
            className="sc-gzVnrw cVeLgT"
          >
            <a
              className="sc-bdVaJa gASuaN"
              href="https://news.ycombinator.com/login?goto=news"
            >
              login
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </td>
</tr>
`;
